src/integration.test.ts

```diff
@@ -16,7 +16,7 @@ const getKey = (paths: string[], key: string): string => {
 };
 
 describe("Integration Test", (): void => {
-  const EXEC_OPTIONS = { shell: "/usr/bin/bash" };
+  const EXEC_OPTIONS = { shell: "/bin/sh" };
```

Comment:
I can see that [macos-12 ships with Bash](https://github.com/actions/runner-images/blob/17295fdcdeefbcac13560031c8ee222d84085bd3/images/macos/macos-12-Readme.md). Maybe try running `whereis bash` and/or `which bash` to ask for the appropriate path on each OS. You can do this by adding a step to the Test workflow before the point at which it's crashing.